packages/cli/source/manifest/compiler.ts

```diff
@@ -1,33 +1,60 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
-/* eslint-disable @typescript-eslint/no-explicit-any */
+import { App, Manifest, Query, Resource } from "@hypertool/common";
 
 import yaml from "js-yaml";
 import fs from "fs";
 import path from "path";
+import joi from "joi";
 import chalk from "chalk";
+import { constants } from "@hypertool/common";
+import lodash from "lodash";
 
-import { paths, getMissingKeys, constants } from "../utils";
-import type { Manifest, App, Query, Resource } from "../types";
-
-const IDENTIFIER_REGEX = /^[a-zA-Z_][a-zA-Z_0-9-]+[a-zA-Z_0-9]$/;
+import { paths } from "../utils";
 
 let errorCount = 0;
 
-export const logMissingKeys = (
-    regsitryType: string,
-    keys: string[],
-    filePath = "<anonymous>",
-) => {
-    errorCount++;
-    console.log(
-        `${chalk.red(
-            "[error]",
-        )} ${filePath}: The following keys are missing in ${regsitryType}: ${keys.join(
-            ", ",
-        )}\n`,
-    );
-};
+const IDENTIFIER_REGEX = /^[a-zA-Z_][a-zA-Z_0-9-]+[a-zA-Z_0-9]$/;
```

Comment:
There is already a constant for entity names in `@hypertool/common`. It's called `namePattern`.